tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -313,6 +313,25 @@ def test_copy_object_acl(self, s3_bucket):
             assert ((response['Grants'][0]['Permission'] == 'FULL_CONTROL') and
                     (len(response['Grants']) == 1))
 
+    @mock_s3
+    def test_delete_bucket(self, s3_bucket):
+        # assert if the bucket is created
+        mock_hook = S3Hook()
+        mock_hook.create_bucket(bucket_name=s3_bucket)
+        assert mock_hook.check_for_bucket(bucket_name=s3_bucket)
+        # add keys to bucket
+        for i in range(0, 3):
+            mock_hook.load_file_obj(
```

Comment:
You should assert for the mock call - not call the mock here. `mock_hook.load_string.assert_called_once_with`

Reply:
mock_hook.load_string is not getting called, this is just to load the data so that the delete function can be tested

Reply:
Actually, I would remove this code / the for-loop, because that is what you would test in the system test not the unit test. The unit is the delete operation only.

Because you are mocking here you don't need to "load"  data to s3 - it won't be loaded anyway.